contract/src/main.rs

```diff
@@ -800,67 +793,34 @@ pub extern "C" fn set_approval_for_all() {
 
     let caller: Key = utils::get_verified_caller().unwrap_or_revert();
 
-    let caller_dictionary_item_key = utils::get_owned_tokens_dictionary_item_key(caller);
-
     let approved_uref = utils::get_uref(
         OPERATOR,
         NFTCoreError::MissingStorageUref,
         NFTCoreError::InvalidStorageUref,
     );
 
-    let identifier_mode: NFTIdentifierMode = utils::get_stored_value_with_user_errors::<u8>(
-        IDENTIFIER_MODE,
-        NFTCoreError::MissingIdentifierMode,
-        NFTCoreError::InvalidIdentifierMode,
-    )
-    .try_into()
-    .unwrap_or_revert();
-
-    let maybe_owned_tokens: Option<Vec<TokenIdentifier>> = match identifier_mode {
-        NFTIdentifierMode::Ordinal => utils::get_dictionary_value_from_key::<Vec<u64>>(
-            OWNED_TOKENS,
-            &caller_dictionary_item_key,
-        )
-        .map(|token_hashes| {
-            token_hashes
-                .into_iter()
-                .map(TokenIdentifier::new_index)
-                .collect()
-        }),
-        NFTIdentifierMode::Hash => utils::get_dictionary_value_from_key::<Vec<String>>(
-            OWNED_TOKENS,
-            &caller_dictionary_item_key,
-        )
-        .map(|token_hashes| {
-            token_hashes
-                .into_iter()
-                .map(TokenIdentifier::new_hash)
-                .collect()
-        }),
-    };
+    let owned_tokens = utils::get_all_owned_token_identifiers(caller);
 
-    if let Some(owned_tokens) = maybe_owned_tokens {
-        // Depending on approve_all we either approve all or disapprove all.
-        for token_id in owned_tokens {
-            // We assume a burnt token cannot be approved
-            if utils::is_token_burned(&token_id) {
-                runtime::revert(NFTCoreError::PreviouslyBurntToken)
-            }
-            if approve_all {
-                storage::dictionary_put(
-                    approved_uref,
-                    &token_id.get_dictionary_item_key(),
-                    Some(operator),
-                );
-            } else {
-                storage::dictionary_put(
-                    approved_uref,
-                    &token_id.get_dictionary_item_key(),
-                    Option::<Key>::None,
-                );
-            }
+    // Depending on approve_all we either approve all or disapprove all.
+    for token_id in owned_tokens {
+        // We assume a burnt token cannot be approved
+        if utils::is_token_burned(&token_id) {
+            runtime::revert(NFTCoreError::PreviouslyBurntToken)
         }
-    };
+        if approve_all {
+            storage::dictionary_put(
+                approved_uref,
+                &token_id.get_dictionary_item_key(),
+                Some(operator),
+            );
+        } else {
+            storage::dictionary_put(
+                approved_uref,
+                &token_id.get_dictionary_item_key(),
+                Option::<Key>::None,
+            );
+        }
+    }
 }
```

Comment:
maybe do `let value = Some / None ... `, followed by 
```
storage::dictionary_put(
                approved_uref,
                &token_id.get_dictionary_item_key(),
                value
            );
```

Reply:
Refactored as per the suggestion